src/box/tuple_format.c

```diff
@@ -1155,17 +1155,23 @@ tuple_field_map_create_plain(struct tuple_format *format, const char *tuple,
 	}
 	defined_field_count = MIN(defined_field_count,
 				  tuple_format_field_count(format));
+	/*
+	 * At this point, if exact_field_count is set, then all required_fields
+	 * are definitely present.
+	 */
+	bool validate_required_fields =
+		validate && format->exact_field_count == 0;
```

Comment:
`exact_field_count` may actually be less than required field count:

```lua
s = box.schema.create_space('test', {field_count = 1})
s:create_index('pk', {parts = {2, 'unsigned'}})
```

Reply:
Thanks for catching this, reworked the last patch. Added this testcase to `test/box/alter.test.lua`.
Also updated `test/box/ddl_collation.test.lua`, because now required fields are checked before field types.